From 169627b77012a3c8368d68fea0b44dde66c3bd85 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Wed, 28 Aug 2024 19:59:54 -0400
Subject: [PATCH] Ensure that ENOSYS is defined even for older kernel

---
 lsfd-cmd/lsfd.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lsfd-cmd/lsfd.c b/lsfd-cmd/lsfd.c
index 034cd3ac7..57429504e 100644
--- a/lsfd-cmd/lsfd.c
+++ b/lsfd-cmd/lsfd.c
@@ -72,6 +72,10 @@ static int kcmp(pid_t pid1 __attribute__((__unused__)),
  * Defined in linux/include/linux/sched.h private header file. */
 #define PF_KTHREAD		0x00200000	/* I am a kernel thread */
 
+#ifndef ENOSYS
+#define ENOSYS 38
+#endif
+
 #include "c.h"
 #include "list.h"
 #include "closestream.h"
-- 
2.43.0
