From f9fdb4c5210abcf97c19a6c76c2e9cd171a484b3 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Wed, 28 Aug 2024 22:06:50 -0400
Subject: [PATCH 2/2] Disable building liblastlog2 on osx

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 6d1f986af..2ccab5d95 100644
--- a/meson.build
+++ b/meson.build
@@ -92,7 +92,7 @@ build_libuuid = not get_option('build-libuuid').disabled()
 conf.set('HAVE_LIBUUID', build_libuuid ? 1 : false)
 summary('libuuid', build_libuuid ? 'enabled' : 'disabled', section : 'components')
 
-build_liblastlog2 = not get_option('build-liblastlog2').disabled()
+build_liblastlog2 = not get_option('build-liblastlog2').disabled() and not (host_machine.system() == 'darwin')
 conf.set('HAVE_LIBLASTLOG2', build_liblastlog2 ? 1 : false)
 summary('liblastlog2', build_liblastlog2 ? 'enabled' : 'disabled', section : 'components')
 
-- 
2.43.0

